corehq/apps/locations/forms.py

```diff
@@ -241,6 +246,12 @@ def clean_site_code(self):
     def clean_location_type(self):
         loc_type = self.cleaned_data['location_type']
 
+        try:
+            loc_type_obj = LocationType.objects.get(Q(code=loc_type)|Q(name=loc_type))
```

Comment:
You're thinking of Locations, it looks like we don't actually enforce *any* uniqueness constraints on `LocationType`s...
That seems like a problem.  I just checked on prod, and it looks possible to add a unique_together on both `('domain', 'name')` and `('domain', 'code')`.  There's only one real domain that needs to be modified, and it looks like a mistake that there's an issue anyways.
```python
(LocationType.objects
 .values('domain', 'code')
 .order_by()
 .annotate(count=Count('pk'))
 .filter(count__gt=1))
```

As for this immediate issue, once Sravan releases the new bulk upload feature, `LocationForm` will no longer be used in bulk upload, and we can switch this to the unambiguous `location_type_pk`.

My preference for actions to take here are to leave this as-is, immediately add those `unique_together`s, and once we're no longer using this in bulk upload, switch to pk references.  How does that sound?

Sheel and I were also just talking about how the CRUD workflow around Locations needs a lot of love, and I think we'll soon do a redesign and rewrite.